spec/models/game_spec.rb

```diff
@@ -43,4 +43,50 @@
       expect(game.white_to_move).to eq(true)
     end
   end
+
+  describe "Populate Board" do
+    it "should have correct number of pieces" do
+      expect(game.pieces.count).to eq 32
+    end
+
+    it "should be empty on the squares" do
+      expect(game.pieces.where(rank: 3..6)).to eq([])
+    end
+
+    (1..8).each do |file|
+      it "should have a white pawn at rank 2, file #{file}" do
+        white_pawn = game.pieces.find_by(rank: 2 , file: file)
```

Comment:
I know I totally suggested using `find_by` here, but it now occurs to me that `where` is a better approach because you can then check the count of the result set to make sure that there aren't multiple pieces on the square in question. `find_by` only returns the first result that matches the given conditions, so it'll mask this scenario. Here's an example of the new approach:
```
pieces = game.pieces.where(rank: 2, file: file)
expect(pieces.count).to eq(1)
# implement other tests using pieces.first
```
Sorry about not mentioning this earlier. Writing thorough tests is pretty tough. There are so many scenarios to consider.